src/device/generate.py

```
@@ -348,124 +373,54 @@ def custom_sort_key(fn):
 
   for fn in primary_funcs:
     sym = paste("_", "ncclDevFunc", *fn)
-    if fn[2] == "LL128":
-      out("#if (defined(__gfx90a__) || defined(__gfx942__) || defined(__gfx950__)) && defined(ENABLE_LL128)\n")
-      out("%s %s();\n#else\n" % (func_declaration, sym))
-      fn_ll = fn[:2] + ("LL",) + fn[3:]
-      sym_ll = paste("_", "ncclDevFunc", *fn_ll)
-      out("%s %s();\n#endif\n" % (func_declaration, sym_ll))
+    guard = get_arch_guard(fn)
+    if guard:
+      out("#if %s\n%s %s();\n#endif\n" % (guard, func_declaration, sym))
     else:
       out("%s %s();\n" % (func_declaration, sym))
   out("\n")
 
+  index = {val: 0 for val in all_unrolls}
```

Would you consider having the default be something other than zero?  I think this may be prone to silently failing on changes.  None is probably safer and then force initializing to zero.  Not a big deal, but I thought that I'd mention it.